lib/utils/request.ts

```diff
@@ -0,0 +1,68 @@
+import { request as nodeHttpRequest } from "http";
+import { request as nodeHttpsRequest } from "https";
+import {
+  supportsSendBeacon,
+  supportsXMLHttpRequest,
+  supportsNodeHttpModule
+} from "./featureDetection";
+
+type RequestType = (url: string, data: object) => void;
+
+const request: RequestType = (url, data) => {
+  const fn = makeRequester();
+  fn(url, data);
+};
+
+function makeRequester() {
+  if (supportsSendBeacon()) {
+    return requestWithSendBeacon;
+  }
+
+  if (supportsXMLHttpRequest()) {
+    return requestWithXMLHttpRequest;
+  }
+
+  if (supportsNodeHttpModule()) {
+    return requestWithNodeHttpModule;
```

Comment:
if we dynamically require('https') here, I think there might be less of an effect of having the module